src/torcontrol.cpp

```diff
@@ -292,10 +297,48 @@ static std::map<std::string,std::string> ParseTorReplyMapping(const std::string
             if (ptr == s.size()) // unexpected end of line
                 return std::map<std::string,std::string>();
             ++ptr; // skip closing '"'
-            /* TODO: unescape value - according to the spec this depends on the
-             * context, some strings use C-LogPrintf style escape codes, some
-             * don't. So may be better handled at the call site.
+            /**
+             * Escape value. Per https://spec.torproject.org/control-spec section 2.1.1:
+             *
+             *   For future-proofing, controller implementors MAY use the following
+             *   rules to be compatible with buggy Tor implementations and with
+             *   future ones that implement the spec as intended:
+             *
+             *     Read \n \t \r and \0 ... \377 as C escapes.
+             *     Treat a backslash followed by any other character as that character.
              */
+            std::string escaped_value;
+            for (size_t i = 0; i < value.size(); ++i) {
+                if (value[i] == '\\') {
+                    ++i;
+                    if (value[i] == 'n') {
```

Comment:
It's not a potential out-of-bounds read, because this is operating on the whole parsed QuotedString. If the last character in the QuotedString was a `\` then the parser would have already detected it and returned earlier while it was trying to find the end of the QuotedString (because the `\` would be interpreted as a `\"`). I'll add a comment to this effect.